crates/provider/src/provider/get_block.rs

```diff
@@ -439,6 +441,127 @@ impl<N: alloy_network::Network> SubFullBlocks<N> {
     }
 }
 
+
+#[derive(Debug)]
+#[must_use = "this does nothing unless you call `.into_stream`"]
+#[cfg(feature = "pubsub")]
+pub struct SubFinalizedBlocks<N: alloy_network::Network> {
+    sub: super::GetSubscription<(SubscriptionKind,), N::HeaderResponse>,
+    client: alloy_rpc_client::WeakClient,
+    kind: BlockTransactionsKind,
+    last_finalized: Arc<tokio::sync::Mutex<Option<N::HeaderResponse>>>,
+}
+
+
+#[cfg(feature = "pubsub")]
+impl<N: alloy_network::Network> SubFinalizedBlocks<N> {
+    /// Create a new [`SubFullBlocks`] subscription with the given [`super::GetSubscription`].
+    ///
+    /// By default, this subscribes to block with tx hashes only. Use [`SubFullBlocks::full`] to
+    /// subscribe to blocks with full transactions.
+    pub fn new(
+        sub: super::GetSubscription<(SubscriptionKind,), N::HeaderResponse>,
+        client: alloy_rpc_client::WeakClient,
+    ) -> Self {
+        Self { sub, client, kind: BlockTransactionsKind::Hashes, last_finalized: Arc::new(tokio::sync::Mutex::new(None)) }
+    }
+
+    /// Subscribe to blocks with transaction hashes only.
+    pub const fn hashes(mut self) -> Self {
+        self.kind = BlockTransactionsKind::Hashes;
+        self
+    }
+
+    /// Set the channel size
+    pub fn channel_size(mut self, size: usize) -> Self {
+        self.sub = self.sub.channel_size(size);
+        self
+    }
+
+    pub fn last_finalized(&self) -> Arc<tokio::sync::Mutex<Option<N::HeaderResponse>>> {
+        self.last_finalized.clone()
+    }
+
+    /// Subscribe to the inner stream of headers and map them to block responses.
+    pub async fn into_stream(
+        self,
+    ) -> TransportResult<impl Stream<Item=TransportResult<N::BlockResponse>> + Unpin> {
```

Comment:
can we introduce a dedicated stream type for this instead?

impl Stream makes it a bit harder to add additional logic later